src/test/llmtemplate_test.cpp

```diff
@@ -0,0 +1,332 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include <continuous_batching_pipeline.hpp>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <openvino/openvino.hpp>
+#include <pybind11/embed.h>
+
+#include "../llm/http_payload.hpp"
+#include "../llm/llm_executor.hpp"
+#include "../llm/llmnoderesources.hpp"
+#include "../mediapipe_internal/mediapipegraphdefinition.hpp"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#include "mediapipe/framework/calculator_runner.h"
+#pragma GCC diagnostic pop
+
+#include "opencv2/opencv.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+class LLMChatTemplateTest : public TestWithTempDir {
+private:
+    std::string tokenizerFilePath;
+    std::string jinjaFilePath;
+
+    void CreateConfig(std::string& fileContents, std::string& filePath) {
+        std::ofstream configFile{filePath};
+        SPDLOG_INFO("Creating config file: {}\n with content:\n{}", filePath, fileContents);
+        configFile << fileContents << std::endl;
+        configFile.close();
+        if (configFile.fail()) {
```

Comment:
Is there a reason we should proceed with a test when the writing fails? I think we should return error from the file (bool should be enough) and assert that when running a test